exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -323,11 +324,16 @@ func addSingleNumberDataPoint(pt pdata.NumberDataPoint, resource pdata.Resource,
 		// convert ns to ms
 		Timestamp: convertTimeStamp(pt.Timestamp()),
 	}
-	switch pt.Type() {
-	case pdata.MetricValueTypeInt:
-		sample.Value = float64(pt.IntVal())
-	case pdata.MetricValueTypeDouble:
-		sample.Value = pt.DoubleVal()
+	// check for staleness marker flag
+	if pt.Flags().HasFlag(pdata.MetricDataPointFlagNoRecordedValue) {
+		sample.Value = math.Float64frombits(value.StaleNaN)
+	} else {
+		switch pt.Type() {
+		case pdata.MetricValueTypeInt:
+			sample.Value = float64(pt.IntVal())
```

Comment:
Not in this PR, but do we ever do this? I am sure that only Prometheus receiver does this stale marker, and they don't use the int.

Reply:
The switch statement is for the case when the datapoint has no OTLP flag, which means the data is not a staleness marker. I changed it to enhance the readability.
 
I also checked the `prometheusreceiver` part and there is no usage that the metric value type is integer at this moment, but should we leave this `switch-case` in case of we have the integer values from OTLP in the future? I found some test cases like [uint64_counter](https://github.com/OpenObservability/OpenMetrics/tree/main/tests/testdata/parsers/uint64_counter)